src/server/cluster/cluster_family.cc

```diff
@@ -1088,6 +1088,70 @@ size_t ClusterFamily::MigrationsErrorsCount() const {
   return error_num;
 }
 
+void ClusterFamily::ReconcileMasterFlow() {
+  auto config = cluster::ClusterConfig::Current();
+
+  for (auto& info : config->GetMutableConfig()) {
+    if (info.master.id == id_) {
+      if (!info.replicas.empty()) {
+        LOG_IF(ERROR, info.replicas.size() > 1)
+            << "More than one replica found, slot redirection after takeover corrupted";
+
+        // assumes there is one replica per master node
+        // TODO figure a smart way to get the replica id_ so
+        // we can find it here
+        info.master = info.replicas.front();
```

Comment:
It's safe to assume that there is one replica. When this changes we should also take care of this. Furthermore, we need to figure out a smart way to the replica.id after `DflyCmd::TakeVer`.

@romange let me know if you disagree